colab_evaluation_test.py

```diff
@@ -0,0 +1,167 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for module colab_evaluation.py."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import gzip
+import os
+import time
+import math
+
+from absl import flags
+from absl.testing import absltest
+from absl.testing import parameterized
+import numpy as np
+import pandas as pd
+import colab_evaluation
+import inference
+import tensorflow.compat.v1 as tf
+
+FLAGS = flags.FLAGS
+
+
+class ColabEvaluationTest(parameterized.TestCase):
+    def setUp(self):
+        super(ColabEvaluationTest, self).setUp()
+
+    def test_parse_sharded_inference_results(self):
+        def generate_random_inferences(n):
+            serialized_inferences = []
+            accessions_list = []
+            activations_list = []
+
+            for _ in range(n):
+                accession = f"ACCESSION_{time.time()}"
+                activations = np.random.rand(100)
+                accessions_list.append(accession)
+                activations_list.append(activations)
+                serialized_inferences.append(
+                    inference.serialize_inference_result(
+                        accession, activations))
+
+            return serialized_inferences, accessions_list, activations_list
+
+        # Create input inference results.
+        num_examples = 100
+        batch_size = 9
+
+        serialized_inferences, accessions_list, activations_list = generate_random_inferences(
+            num_examples)
+
+        shard_1_contents = b"\n".join(serialized_inferences[0:60])
+        shard_2_contents = b"\n".join(serialized_inferences[60:])
+
+        shard_dir = self.create_tempdir()
+
+        shard_1_filename = shard_dir.create_file('shard_1').full_path
+        shard_2_filename = shard_dir.create_file('shard_2').full_path
+
+        # Write contents to a gzipped file.
+        with tf.io.gfile.GFile(shard_1_filename, 'wb') as f:
+            with gzip.GzipFile(fileobj=f, mode='wb') as f_gz:
+                f_gz.write(shard_1_contents)
+
+        with tf.io.gfile.GFile(shard_2_filename, 'wb') as f:
+            with gzip.GzipFile(fileobj=f, mode='wb') as f_gz:
+                f_gz.write(shard_2_contents)
+
+        # Read these shards.
+
+        iterator = colab_evaluation.batched_inferences_from_dir(
+            shard_dir.full_path, batch_size=batch_size)
+
+        actual = list(iterator)
+
+        # Check output.
+
+        self.assertEqual(len(actual), math.ceil(num_examples / batch_size))

```

Comment:
The tested function by design returns batched results - here we check that the batches make sense. The name of this function in the test was out of date so that was unclear, now updated. Thanks.